infrastructure/helm-chart/charts/core/charts/upgrading/templates/0.29.0.yaml

```diff
@@ -0,0 +1,283 @@
+        webhooks=$(cat /tmp/response.txt | head -n 1)
+        if [ -z "${webhooks}" ]; then
+          kubectl patch configmap upgrade-{{ .Values.global.kubernetes.appImageTag }} --type merge -p '{"data":{"webhookUsed": "false"}}'
+        else
+          kubectl patch configmap upgrade-{{ .Values.global.kubernetes.appImageTag }} --type merge -p '{"data":{"webhookUsed": "true"}}'
+          echo ${webhooks} > /tmp/webhooks.yaml
+          kubectl delete configmap existing-webhooks
+          sleep 1
+          kubectl create configmap existing-webhooks --from-file=/tmp/webhooks.yaml
+          kubectl label configmap existing-webhooks core.airy.co/upgrade="true"
+        fi
+      fi
+    fi
+
+  subscribe-wehoook.sh: |
+    #!/bin/sh
+    system_token=${SYSTEM_TOKEN}
+    webhookUsed=$(kubectl get configmap upgrade-0.29.0-alpha -o jsonpath='{.data.webhookUsed}')
+    echo "Webhook used: ${webhookUsed}"
+    if [ "${webhookUsed}" = "true" ]; then
+      request=$(kubectl get configmap existing-webhooks -o jsonpath='{.data.webhooks\.yaml}')
+      curl -X POST -H "Content-Type: application/json" -H "Authorization: Bearer ${system_token}" -d "${request}" api-admin/webhooks.subscribe
+    fi
+---
+# Cleanup old jobs
+apiVersion: batch/v1
+kind: Job
+metadata:
+  name: {{ .Values.global.kubernetes.appImageTag }}-cleanup-upgrade
+  namespace: {{ .Values.global.kubernetes.namespace }}
+  annotations:
+    "helm.sh/hook": "pre-upgrade"
+    "helm.sh/hook-weight": "0"
+  labels:
+    core.airy.co/upgrade: "true"
+spec:
+  ttlSecondsAfterFinished: 120
+  template:
+    spec:
+      containers:
+      - name: reset-app
+        image: "bitnami/kubectl"
+        command: ['/bin/sh', '/opt/provisioning/upgrade-cleanup.sh']
+        volumeMounts:
+        - name: upgrading-scripts
+          mountPath: /opt/provisioning
+      serviceAccountName: airy-upgrade
+      volumes:
+        - name: upgrading-scripts
+          configMap:
+            name: upgrading-scripts
+      restartPolicy: Never
+  backoffLimit: 3
+---
+# Migrate webhooks
+apiVersion: batch/v1
+kind: Job
+metadata:
+  name: {{ .Values.global.kubernetes.appImageTag }}-upgrade-integration-webhooks
+  namespace: {{ .Values.global.kubernetes.namespace }}
+  annotations:
+    "helm.sh/hook": "pre-upgrade"
+    "helm.sh/hook-weight": "10"
+  labels:
+    core.airy.co/upgrade: "true"
+spec:
+  ttlSecondsAfterFinished: 120
+  template:
+    spec:
+      initContainers:
+      - name: wait-api-admin
+        image: busybox
+        command: ["/bin/sh", "/opt/provisioning/wait-for-service.sh"]
+        env:
+        - name: SERVICE_NAME
+          value: api-admin
+        - name: SERVICE_PORT
+          value: "80"
+        volumeMounts:
+        - name: provisioning-scripts
+          mountPath: /opt/provisioning
+      - name: check-webhook
+        image: bitnami/kubectl
+        command:  ['/bin/sh', '/opt/provisioning/check-existing-wehook.sh']
+        env:
+        - name: SYSTEM_TOKEN
```

Comment:
Perhaps a good case for
```yaml
        envFrom:
          - configMapRef:
              name: security
```

since not all (e.g. minikube) installations have a system token.

Reply:
Good catch! :+1: I will change ghat.